algorithms/ar-msrt/typescript/merge_sort.ts

```diff
@@ -0,0 +1,30 @@
+function mergeSort(array: number[]) : number[]{
+    const merge = function (array1 : number[], array2 : number[]) : number[] {
+        if (array1 == [] || array1 == undefined){
+            return array2;
+        } else if (array2 == [] || array2 == undefined){
+            return array1;
+        } else {
+            let output: number[] = [];
+            while(array1.length > 0 || array2.length > 0){
+                if (array1.length == 0) {
+                    output.push(array2.shift());
+                } else if (array2.length == 0) {
```

Comment:
Codacy found an issue: [Expected '===' and instead saw '=='.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6334421)